docker/registry.Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM ubuntu:20.04
```

Comment:
Maybe you can add some comments why you need certain things. I bet in half a year you don't remember:
* Why you need this user mahiru and a home folder
* Why you need the `/var/log` and `/var/run` folders

Reply:
Well, running services as root is generally bad practice, so you see this making a user in every Dockerfile. I guess I could have gone with `/opt/mahiru` instead of `/home/mahiru` but it doesn't really matter. Logging is very useful for debugging and `/var/log` is where logs live according to the [FHS](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/index.html), and `/var/run` is where you put your pid files according to same. I looked at how I [did this for Cerise](https://github.com/MD-Studio/cerise/blob/develop/conf/docker-init.sh) two years ago and still remembered how and why, so I think I'll be okay in six months.

I think anyone familiar with Docker files and basic GNU/Linux system administration will be able to figure this one out, it's nothing special.

Reply:
Maybe this is something we need to discuss with the team actually, but I like this quote from Google's Python style guide: "On the other hand, never describe the code. Assume the person reading the code knows Python (though not what you’re trying to do) better than you do." Writing tutorials are fine, but it should be kept separate from programming.

Reply:
I totally agree to never describe the code. But *why* you need code is different from that.

I don't think you can assume that all future developers will know all best practices for docker. I never created users in my docker images nor home folders, probably because it makes things overly complicated and I am lazy. That's why I was wondering whether there was a particular reason for it.

But feel free to leave it like this!

Reply:
I don't think the point of that statement is to point out that everyone at Google is an expert at Python. There has to be *someone* at Google who is their least-experienced Python developer, and who may therefore not be able to immediately understand the code their colleagues are writing.

The point is that if you require the programmer to document everything that may be unknown to someone, they'll end up starting every docstring with "Python is a programming language. Programming languages are used to ...". There's no reasonable limit, and you end up drowning the code in explanations that make the code harder to read for more experienced developers, all to support a beginner programmer who may never actually exist.

Clearly, if a less experienced programmer needs to work on the code, then they may run into language features or idiom or algorithms that they don't understand. And then, in order to be able to do their job, they're going to have to learn about this unknown thing. But they should do so by looking up a tutorial or a blog post or the documentation, and those should be separate objects. The purpose of code is not to teach people how to program (unless it was written for a tutorial), and trying to tack on that responsibility will make the code less clear.

So, as a specific example, if you saw this Dockerfile and you wondered why it didn't just run as root, then you could do a web search and find [this blog post](https://medium.com/@mccode/processes-in-containers-should-not-run-as-root-2feae3f0df3b) that explains things much better than any comment could.

Reply:
I think these are all good points. I do think that just putting a small comment doesn't really do harm, and can often clarify a lot and prevent people for having to search context to understand code. But let's not discuss the universe on this PR and let's agree to 90% agree and 10% disagree ;)

Reply:
Nice blog post by the way :)